ledger-api/grpc-definitions/com/daml/ledger/api/v1/event.proto

```diff
@@ -50,14 +52,30 @@ message CreatedEvent {
   // Required
   Identifier template_id = 3;
 
-  // The key of the created contract, if defined.
+  // The key of the created contract.
+  // This will be set if and only if ``create_arguments`` is set and ``template_id`` defines a contract key.
   // Optional
   Value contract_key = 7;
 
   // The arguments that have been used to create the contract.
-  // Required
+  // Set either:
+  //   - if there was a party, which is in the ``witness_parties`` of this event,
+  //     and for which an ``InclusiveFilters`` exists with the ``template_id`` of this event
+  //     among the ``template_ids``,
+  //   - or if there was a party, which is int the ``witness_parties`` of this event,
+  //     and for which a wildcard filter exists (``Filters`` without ``InclusiveFilters``,
+  //     or with an ``InclusiveFilters`` with empty ``template_ids`` and empty ``interface_filters``).
+  // Optional
   Record create_arguments = 4;
 
+  // Interface views specified in the transaction filter.
+  // Includes an ``InterfaceView`` for each interface for which, there is a ``InterfaceFilter`` with
+  //   - its party in the ``witness_parties`` of this event,
+  //   - and which is implemented by the template of this event,
```

Comment:
```suggestion
  //   - and which is implemented by the template of this event at the offset of this event
```

Reply:
That's not correct. Consider this scenario:
1. At offset O1, we have a create event E for template T.
2. At offset O2 > O1, a package gets uploaded that adds an interface I with a retroactive instance for T.
3. Now I send a subscription from ledger begin for I. That will include E.